File: Shared/Exceptions.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2009-2015 ApexRMS.
'
'************************************************************************************

Imports System.Runtime.Serialization

''' <summary>
''' The STSim Exception
''' </summary>
''' <remarks></remarks>
<Serializable()>
Public NotInheritable Class STSimException
    Inherits Exception

    Public Sub New()
        MyBase.New("STSim exception")
    End Sub

    Public Sub New(ByVal message As String)
        MyBase.New(message)
    End Sub

    Public Sub New(ByVal message As String, ByVal innerException As Exception)
        MyBase.New(message, innerException)
    End Sub

    Private Sub New(ByVal info As SerializationInfo, ByVal context As StreamingContext)
        MyBase.New(info, context)
    End Sub

End Class

<Serializable()>
Public NotInheritable Class STSimMapDuplicateItemException
    Inherits Exception

    Public Sub New()
        MyBase.New("Duplicate Item Exception")
    End Sub

    Public Sub New(ByVal message As String)
        MyBase.New(message)
    End Sub

    Public Sub New(ByVal message As String, ByVal innerException As Exception)
        MyBase.New(message, innerException)
    End Sub

    Private Sub New(ByVal info As SerializationInfo, ByVal context As StreamingContext)
        MyBase.New(info, context)
    End Sub

End Class

File: Shared/Exceptions.cs
// ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System;
using System.Runtime.Serialization;

namespace SyncroSim.STSim
{
    [Serializable()]
    public sealed class STSimException : Exception
    {
        public STSimException() : base("STSim exception")
        {
        }

        public STSimException(string message) : base(message)
        {
        }

        public STSimException(string message, Exception innerException) : base(message, innerException)
        {
        }

        private STSimException(SerializationInfo info, StreamingContext context) : base(info, context)
        {
        }
    }

    [Serializable()]
    public sealed class STSimMapDuplicateItemException : Exception
    {
        public STSimMapDuplicateItemException() : base("Duplicate Item Exception")
        {
        }

        public STSimMapDuplicateItemException(string message) : base(message)
        {
        }

        public STSimMapDuplicateItemException(string message, Exception innerException) : base(message, innerException)
        {
        }

        private STSimMapDuplicateItemException(SerializationInfo info, StreamingContext context) : base(info, context)
        {
        }
    }
}
